ext/oci8/oci8_statement.c

```diff
@@ -1767,7 +1772,7 @@ php_oci_bind *php_oci_bind_array_helper_string(zval *var, zend_long max_table_le
 
 	for (i = 0; i < bind->array.current_length; i++) {
 		if ((entry = zend_hash_get_current_data(hash)) != NULL) {
-			convert_to_string_ex(entry);
+			ZEND_ASSERT(Z_TYPE_P(entry) == IS_STRING);
```

Comment:
Why is this an `assert` versus, say, breaking out of the loop and returning error?

Reply:
This function is implemented in two loops: The first one converts everything to string and gathers the maxlength, while the second one does the main processing. After the first loop has run, we're guaranteed that everything we're working with are strings. The assertion is there to document that.

Reply:
Right, but isn't that first conversion loop only run when maxlength is `-1`, meaning the guarantee of stringiness is conditional upon the given value of `maxlength`?

Reply:
Ah you're right, I glossed over that bit. So yeah, these string conversions should stay, or the loop above needs to be run unconditionally.

Reply:
My gut says this is called, in the overwhelming majority, with maxlength = -1, so I'd be in favor of making the loop unconditional anyway. But I balance that with the run cost of walking the loop unnecessarily and wonder if it's more expensive than the double cost of conversion. I don't know, but my gut says conversion is going to be more expensive than the walk-and-access loop.

Reply:
I went with the conservative approach here and just check try_convert_to_string every time. I think it would definitely be better to make the loop unconditional, but just dropping the condition would change the logic subtly (it could further reduce an explicitly passed maxlength), so I think it's safer not to touch it, as I don't have any familiarity with this code.